backend/src/components/auth.js

```diff
@@ -4,14 +4,26 @@ const jsonwebtoken = require('jsonwebtoken');
 const log = require('npmlog');
 const qs = require('querystring');
 
-const {
-  acronymService,
-  userService
-} = require('../services');
+const { acronymService, userService } = require('../services');
 const permissionHelpers = require('./permissionHelpers');
 const utils = require('./utils');
 
 const auth = {
+  /**
+   *  @function getUserAcronyms
+   *  @param {object} req An Express request object
+   *  @param {object} _res An Express response object
+   *  @param {function} next A callback function
+   */
+  async getUserAcronyms(req, _res, next) {
+    if (req.user && !auth.isTokenExpired(req.user.jwt)) {
+      const acronyms = await userService.getUserAcronymList(req.user.id);
+      req.user.acronyms = Array.isArray(acronyms) ? acronyms : [];
```

Comment:
Where is this req.user.acronyms used? Still looks like all the work is from permissionHelper.checkAcronymPermission?

Reply:
`getUserAcronyms` is used in the route layer to expose the list of acronyms a user is associated to the /auth/token response. This is needed for subsequent front-end work. The subtle difference with `permissionHelper.checkAcronymPermission` is that instead of offering an array of acronyms, it answers the question "given this token and the acronym you are asking for, are you allowed to do it", whereas `getUserAcronyms` answers the question "what acronyms does this user have access to?".